src/lib/chawathe98_matcher.rs

```diff
@@ -46,35 +46,2262 @@
+                        // Partition
+                        for n1 in &list_n {
+                            for n2 in &list_n {
+                                // Check if n1.parent == n2.parent are the same
+                                if n1.id() != n2.id() {
+                                    //                                    let n1_parent = self.from_arena[m].parent().unwrap();
+                                    //                                    let n2_parent = self.to_arena[n].parent().unwrap();
+                                }
+                            }
+                        }
+                    }
+                        // //// CASE 2.2 //// //
+                        // Figure 17, appendix D
+                        else if list_m[0].id() == self.from_arena.borrow().size() - 1 {
+
+                        }
+                }
+        }
+    }
+
+    /// add to_nodes to vector to be inserted
+    pub fn insert_node(&mut self, node_to_insert: NodeId<ToNodeId>) {
+        self.list_node_inserted.borrow_mut().push(node_to_insert);
+    }
+    /// add from_node to vector to be deleted
+    pub fn delete_node(&mut self, node_to_delete: NodeId<FromNodeId>) {
+        self.list_node_deleted.borrow_mut().push(node_to_delete);
+    }
+    /// Print
+    pub fn print(&self) {
+        let size_from_node = NodeId::new(self.from_arena.borrow().size() - 1);
+        println!("{:?}", size_from_node);
+        let size_to_node = NodeId::<ToNodeId>::new(self.to_arena.borrow().size() - 1);
+        let last_node_from_node = &self.from_arena.borrow()[size_from_node];
+        println!("The Last node From_NODE is --> {:?}", last_node_from_node);
+        let last_node_to_node = &self.to_arena.borrow()[size_to_node];
+        println!("The Last node From_NODE is --> {:?}", last_node_to_node);
+        println!("List Of edges already mapped to");
+        let mut index = 0;
+        let list_edges = self.list_edges.borrow();
+        for edges in list_edges.iter() {
+            println!(" -- {:?} -- ", index);
+
+            println!(
+                "TY        :   From node    -> {:?} || to node -> {:?}",
+                self.from_arena.borrow()[edges.from_node].ty, self.to_arena.borrow()[edges.to_node].ty
+            );
+            println!(
+                "Value     :   From node -> {:?} || to node -> {:?}",
+                self.from_arena.borrow()[edges.from_node].label, self.to_arena.borrow()[edges.to_node].label
+            );
+            println!(
+                "From node -> {:?} || to node -> {:?} || Edge type -> {:?}",
+                edges.from_node, edges.to_node, edges.edge_type
+            );
+            index += 1;
+            println!("--");
+        }
+    }
+}
+
+/// The Chawathe98Config2 matcher needs no configuration.
 #[derive(Debug, Clone, PartialEq)]
-pub struct Chawathe98Config  {}
+pub struct Chawathe98Config2 {}
```

Comment:
It's unfortunate that we end up having two of these configs, but can we call them something other than `Config2`?